deps/spidershim/src/v8isolate.h

```diff
@@ -19,9 +19,49 @@
 // IN THE SOFTWARE.
 
 #pragma once
+
+#include <stack>
+
 #include "v8.h"
+#include "v8context.h"
+#include "rootstore.h"
+#include "mozilla/Maybe.h"
 
 namespace v8 {
 
+struct Isolate::Impl {
+  JSRuntime* rt;
+  std::vector<Context*> contexts;
+  std::stack<Context*> currentContexts;
+  std::vector<StackFrame*> stackFrames;
+  std::vector<StackTrace*> stackTraces;
+  mozilla::Maybe<internal::RootStore> persistents;
+  mozilla::Maybe<internal::RootStore> eternals;
+  std::vector<void *> messageListeners;
```

Comment:
I'm not a huge fan of erasing the type like this.  Any reason why you can't make this a vector of `MessageCallback`s?